### ISSUE TITLE
dynamicTyping attempts to parse string as Date if contains ISO date string

### ISSUE THREAD

**Issue by thebetternewt** [mholt/PapaParse#871]
Original value:
"INITIAL ISSUE - 2021-01-25T21:11:14Z"

Enabling:
`dynamicTyping: true`

Causes the parser to parse the string as a JS Date object, resulting in "Invalid Date".

**Comment by thebetternewt** [comment]
Looks like `ISO_DATE` should be:
```
var ISO_DATE = /^(\d{4}-[01]\d-[0-3]\dT[0-2]\d:[0-5]\d:[0-5]\d\.\d+([+-][0-2]\d:[0-5]\d|Z))|(\d{4}-[01]\d-[0-3]\dT[0-2]\d:[0-5]\d:[0-5]\d([+-][0-2]\d:[0-5]\d|Z))|(\d{4}-[01]\d-[0-3]\dT[0-2]\d:[0-5]\d([+-][0-2]\d:[0-5]\d|Z))$/;
```

**Comment by pokoli** [comment]
hi @thebetternewt, 

Could you please upload a pull request with a test case?
Thanks.

**Comment by pokoli** [comment]
Fixed by #872
